File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        Map<Integer, Integer> map = new HashMap<>();
        int length = nums.length;
        
        for(int i = 0 ; i < length ; i++)
        {
            int rem = target - nums[i];
            
            if(map.containsKey(rem))
                return new int[]{i, map.get(rem)};
            
            map.put(nums[i], i);           
        }
        
        return new int[]{-1, -1};
    }
}

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    
    let map = new Map();
    let length = nums.length;
    
    for( let i = 0 ; i < length ; i++)
    {
        let rem = target - nums[i];
        
        if(map.has(rem))
        {
            return new Array(i, map.get(rem));
        }
        
        map.set(nums[i], i);
    }
    
    return new Array(-1, -1);
};